### PR TITLE
Update DeveloperGuide.md

### LINKED ISSUE #470: [DG] Fix TOC links
Links don't seem to be working currently.

image

### LINKED ISSUE #472: [DG] Fix user stories
- [x] Standardise between 'user' and 'CS2100 TA'
- [x] Fix any improper formats: image

### LINKED ISSUE #473: [DG] Use case bugs
- [x] Point 4: ProgrammerError does not make any request to the user
 image
- [x] UC6: Not a use case? 
image

### LINKED ISSUE #475: [DG] Table of Contents Bug
The table of contents is not clickable to bring to the desired section

### PR DESCRIPTION
**Pull request by erwinqxy**
Fix #470
Fix #472 
Fix #473 
Fix #475

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -270,24 +272,24 @@ The following sequence diagram shows how the undo operation works:
 The `redo` command does the opposite — it calls `Model#redoAddressBook()`, which shifts the `currentStatePointer` once
 to the right, pointing to the previously undone state, and restores the address book to that state.
 
-<div markdown="span" class="alert alert-info">:information_source: **Note:** If the `currentStatePointer` is at index `addressBookStateList.size() - 1`, pointing to the latest address book state, then there are no undone AddressBook states to restore. The `redo` command uses `Model#canRedoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the redo.
+<div markdown="span" class="alert alert-info">:information_source: **Note:** If the `currentStatePointer` is at index `addressBookStateList.size() - 1`, pointing to the latest address book state, then there are no undone AddressBook states to restore. The `redo` command uses `Model#canRedoAddressBook()` to check if this is the case. If so, it will return an error to the CS2100 TA rather than attempting to perform the redo.
```
Comment (AllardQuek) [inline docs/DeveloperGuide.md:275]: Should we change the method names to refer to ProgrammerError?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -334,15 +336,15 @@ the lifeline reaches the end of diagram.>
...
+The following UML activity diagram summarizes what happens when a CS2100 TA executes a new filter command.
 
 #### Design Consideration
 
 **Aspect: How filter command executes**
 
 * **Alternative 1 (current choice):** Filter commands can take in any combination of query parameters (`-n`, `-cid`
   , `-sid`)
-    * Pros: Allow for flexibility in the way users want to filter the list.
+    * Pros: Allow for flexibility in the way the CS2100 TA want to filter the list.
```
Comment (AllardQuek) [inline docs/DeveloperGuide.md:347]: When refactoring, can you make sure the tenses are consistent?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -392,9 +394,9 @@ results of student at index 1 of the current list.
 
 Given below is a possible usage scenario:
 
-[Pre-Condition] There are 2 students in ProgrammerError, and the user has created some lab results for each of them.
+[Pre-Condition] There are 2 students in ProgrammerError, and the CS2100 TA has created some lab results for each of them.
 
-Step 1. The user key in the command `show 1`: The information of the student at index 1 as well as his/her lab results
+Step 1. The CS2100 TA key in the command `show 1`: The information of the student at index 1 as well as his/her lab results
```
Comment (AllardQuek) [inline docs/DeveloperGuide.md:399]: TA "keys"

Similar issue in other areas as well

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (erwinqxy) [reply docs/DeveloperGuide.md:275]: I think we can delete this whole chunk since we don't have an undo / redo operation?

Comment (erwinqxy) [reply docs/DeveloperGuide.md:275]: it will be a fix for #469

Comment (AllardQuek) [reply docs/DeveloperGuide.md:275]: It is a proposed enhancement, which is allowed. We should discuss this with the rest whether to delete this or replace with a different proposed enhancement.

Comment (erwinqxy) [reply docs/DeveloperGuide.md:275]: alright, I will unlink the fix first. other than that should be fine?

Comment (AllardQuek) [reply docs/DeveloperGuide.md:275]: Yup, thank you! 🪂

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -548,11 +551,11 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 | `* *`   | CS2100 TA                                   | use tags to identify which labs are marked or unmarked                                           | know which what labs to mark next.
 | `*`     | CS2100 TA with multiple classes             | filter the contact list by name, classes, email  | easily identify those in the current class.                |                                                                        
 | `*`     | archive previous batch statistics           | compare current batch performance with them                                                      |
```
Comment (AllardQuek) [inline docs/DeveloperGuide.md:553]: Do you want to correct this now too? #473

Comment (erwinqxy) [reply docs/DeveloperGuide.md]: done!

**Review by AllardQuek — APPROVED** [review]
LGTM

**Pull request merged**